simplecpp.cpp

```diff
@@ -392,8 +394,14 @@ void simplecpp::TokenList::readfile(std::istream &istr, const std::string &filen
                     loc.push(location);
                     location.fileIndex = fileIndex(cback()->str.substr(1U, cback()->str.size() - 2U));
                     location.line = 1U;
+                } else if (lastline == "# line %num%") {
+                    loc.push(location);
+                    location.line = atol(cback()->str.c_str());
```

Comment:
please use the portable std::atol(). atol is the old c function that you'll get if you include stdlib.h.